src/Microsoft.DotNet.Darc/DarcLib/VirtualMonoRepo/VmrCodeflower.cs

```diff
@@ -322,6 +323,101 @@ public async Task<LastFlows> GetLastFlowsAsync(
             await DetectCrossingFlow(lastFlow, lastBackflow, lastForwardFlow, repoClone));
     }
 
+    /// <summary>
+    /// Tries to find how many previous flows need to be recreated in order to apply the current changes.
+    /// Iteratively rewinds through previous flows until it finds the one that introduced the conflict with the current changes.
+    /// Then it creates a given branch there and applies the current changes on top of the recreated previous flows.
+    /// </summary>
+    protected async Task RecreatePreviousFlowsAndApplyChanges(
+        SourceMapping mapping,
+        Build build,
+        ILocalGitRepo repo,
+        LastFlows lastFlows,
+        string headBranch,
+        string targetBranch,
+        IReadOnlyCollection<string>? excludedAssets,
+        Func<Task> reapplyChanges,
+        bool currentIsBackflow,
+        CancellationToken cancellationToken)
+    {
+        _logger.LogInformation("Failed to create PR branch because of a conflict. Re-creating previous flows..");
+
+        // Create a fake previously applied build that we will use when reapplying the previous flow.
+        // We only care about the sha here, because it will get overwritten anyway with the current build which will be applied on top.
+        var currentFlowSha = currentIsBackflow ? lastFlows.LastBackFlow!.VmrSha : lastFlows.LastForwardFlow.RepoSha;
+        Build previouslyAppliedBuild = new(-1, DateTimeOffset.Now, 0, false, false, currentFlowSha, [], [], [], [])
+        {
+            GitHubRepository = build.GitHubRepository,
+            AzureDevOpsRepository = build.AzureDevOpsRepository
+        };
+
+        // We recursively try to re-create previous flows until we find the one that introduced the conflict with the current flown
+        int flowsToRecreate = 1;
+        while (flowsToRecreate < 50)
+        {
+            _logger.LogInformation("Trying to recreate {count} previous flow(s)..", flowsToRecreate);
+
+            // We rewing to the previous flow and create a branch there
+            (Codeflow previousFlow, LastFlows previousFlows) = await RewindToPreviousFlowAsync(
+                mapping,
+                repo,
+                flowsToRecreate,
+                lastFlows,
+                headBranch,
+                targetBranch,
+                cancellationToken);
+
+            // We reconstruct the previous flow's branch
```

Comment:
It might be easier to read this method for someone if we say something like
`We replay all previous flows now, excluding manual changes that might have caused the conflict`